sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/README.md

```diff
@@ -1,25 +1,305 @@
+
+#### Using endpoint and token credential
+
+**NOTE:** token credential authentication is supported only for storage tables.
+
+`<ConnectionName>__endpoint` = `https://...table.core.windows.net`
+
+If not credential information is provided the [`DefaultAzureCredential`][identity_dac] is used.
+
+When using user-assigned manageed identity the `clientId` and `credential` settings need to be provided:
+
+`<ConnectionName>__credential` = `managedidentity`
+
+`<ConnectionName>__clientId` = `<user-assigned client id>`
+
+#### Using shared key credential
+
+When using [shared key authentication](https://docs.microsoft.com/rest/api/storageservices/authorize-with-shared-key) the `endpoint`, `accountKey` and `accountName` need to be provided.
+
+`<ConnectionName>__endpoint` = `https://...table.core.windows.net`
+
+`<ConnectionName>__credential__accountName` = `<account name>`
+
+`<ConnectionName>__credential__accountKey` = `<account key>`
 
 ## Key concepts
 
-### Thread safety
+The input binding allows you to read table as input to an Azure Function. The output binding allows you to modify and delete table rows in an Azure Function.
+
+Please follow the [input binding tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table-input?tabs=csharp) and [output binding tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table-output?tabs=csharp) to learn about using this extension for accessing table service.
+
+## Examples
+
+### Binding to a single entity
+
+```C# Snippet:InputSingle
+public class InputSingle
+{
+    [FunctionName("InputSingle")]
+    public static void Run([Table("MyTable", "<PartitionKey>", "<RowKey>")] TableEntity entity, ILogger log)
+    {
+        log.LogInformation($"PK={entity.PartitionKey}, RK={entity.RowKey}, Text={entity["Text"]}");
+    }
+}
+```
+
+### Binding to a single entity using model type
+
+```C# Snippet:MyEntity
+public class MyEntity
+{
+    public string PartitionKey { get; set; }
+    public string RowKey { get; set; }
+    public string Text { get; set; }
+}
+```
+```C# Snippet:InputSingleModel
+public class InputSingleModel
+{
+    [FunctionName("InputSingleModel")]
+    public static void Run([Table("MyTable", "<PartitionKey>", "<RowKey>")] MyEntity entity, ILogger log)
+    {
+        log.LogInformation($"PK={entity.PartitionKey}, RK={entity.RowKey}, Text={entity.Text}");
+    }
+}
+```
+
+### Binding to multiple entities with filter
+
+```C# Snippet:InputMultipleEntitiesFilter
+public class InputMultipleEntitiesFilter
+{
+    [FunctionName("InputMultipleEntitiesFilter")]
+    public static void Run([Table("MyTable", "<PartitionKey>", Filter = "Text ne ''")] IEnumerable<TableEntity> entities, ILogger log)
+    {
+        foreach (var entity in entities)
+        {
+            log.LogInformation($"PK={entity.PartitionKey}, RK={entity.RowKey}, Text={entity["Text"]}");
+        }
+    }
+}
+```
+
+### Creating a single entity
+
```

Comment:
"Output functions are not triggered by table activity, but can modify entities in the table..."